ddpui/api/airbyte_api.py

```diff
@@ -1,748 +1,329 @@
-    orguser: OrgUser = request.orguser
-    if orguser.org.airbyte_workspace_id is None:
-        raise HttpError(400, "create an airbyte workspace first")
-
-    res = airbyte_service.get_sources(orguser.org.airbyte_workspace_id)["sources"]
-    logger.debug(res)
-    return res
-
-
-@airbyte_router.get("/sources/{source_id}", auth=auth.CustomAuthMiddleware())
-@has_permission(["can_view_source"])
-def get_airbyte_source(request, source_id):
-    """Fetch a single airbyte source in the user organization workspace"""
-    orguser: OrgUser = request.orguser
-    if orguser.org.airbyte_workspace_id is None:
-        raise HttpError(400, "create an airbyte workspace first")
-
-    res = airbyte_service.get_source(orguser.org.airbyte_workspace_id, source_id)
-    logger.debug(res)
-    return res
-
-
-@airbyte_router.get("/sources/{source_id}/schema_catalog", auth=auth.CustomAuthMiddleware())
-@has_permission(["can_view_source"])
-def get_airbyte_source_schema_catalog(request, source_id):
-    """Fetch schema catalog for a source in the user organization workspace"""
-    orguser: OrgUser = request.orguser
-    if orguser.org.airbyte_workspace_id is None:
-        raise HttpError(400, "create an airbyte workspace first")
-
-    res = airbyte_service.get_source_schema_catalog(orguser.org.airbyte_workspace_id, source_id)
-    logger.debug(res)
-    return res
-
-
-@airbyte_router.get("/destination_definitions", auth=auth.CustomAuthMiddleware())
-@has_permission(["can_view_warehouses"])
-def get_airbyte_destination_definitions(request):
-    """Fetch destination definitions in the user organization workspace"""
-    orguser: OrgUser = request.orguser
-    if orguser.org.airbyte_workspace_id is None:
-        raise HttpError(400, "create an airbyte workspace first")
-
-    res = airbyte_service.get_destination_definitions(orguser.org.airbyte_workspace_id)[
-        "destinationDefinitions"
-    ]
-    allowed_destinations = os.getenv("AIRBYTE_DESTINATION_TYPES")
-    if allowed_destinations:
-        res = [destdef for destdef in res if destdef["name"] in allowed_destinations.split(",")]
-    logger.debug(res)
-    return res
-
-
-@airbyte_router.get(
-    "/destination_definitions/{destinationdef_id}/specifications",
-    auth=auth.CustomAuthMiddleware(),
-)
-@has_permission(["can_view_warehouse"])
-def get_airbyte_destination_definition_specifications(request, destinationdef_id):
+    try:
+        org: Org = Org.objects.get(id=orgid)
+        airbyte_service: AirbyteService = AirbyteService(org)
+        source_definitions: List[Dict[str, Any]] = airbyte_service.list_source_definitions()
+        return JsonResponse({"source_definitions": source_definitions})
+    except Exception as e:
+        logger.error(f"get_source_definitions: {str(e)}")
+        return JsonResponse({"error": str(e)}, status=500)
+
+def get_source_definition_specifications(request: Request, orgid: str, source_definition_id: str) -> JsonResponse:
     """
```

Comment:
Could we call this `AirbyteSourceDefinitionSpecificationResponse` instead of JsonResponse. Pleas take a look at the issue one more time, i have added some examples there. Should be helpful 

In general, if we could name the schema models to their functionality would be great.